app/src/services/airdrop/index.ts

```diff
@@ -0,0 +1,119 @@
+      this.airdrops = airdrops.map(airdrop => new ethers.Contract(airdrop, airdropAbi, provider).connect(signer))
+    }
+  }
+
+  static getClaim = async (
+    airdrop: string,
+    address: Maybe<string>,
+    networkId: number,
+    provider: any,
+    relay: boolean,
+  ) => {
+    // handle / format address
+    const lowerCaseAddress = address && address.toLowerCase()
+    const recipient = lowerCaseAddress && isAddress(lowerCaseAddress) && getAddress(lowerCaseAddress)
+    // eslint-disable-next-line
+    const proofs = require(`./${airdrop}.json`)
+    if (recipient) {
+      if (networkId === networkIds.XDAI) {
+        const proxyAddress = calcRelayProxyAddress(recipient, provider)
+        const proxyClaim = proxyAddress && proofs.claims[proxyAddress]
+        if (proxyClaim && relay) {
+          return { ...proxyClaim, recipient: proxyAddress }
+        }
+      }
+      const claim = proofs.claims[recipient]
+      if (claim) {
+        return { ...claim, recipient }
+      }
+    }
+  }
+
+  getClaims = async (address: Maybe<string>) => {
+    if (this.airdrops) {
+      const network = await this.provider.getNetwork()
+      const claims = await Promise.all(
+        this.airdrops.map(async airdrop => {
+          const claim = await AirdropService.getClaim(
+            airdrop.address,
+            address,
+            network.chainId,
+            this.provider,
+            this.relay,
+          )
+          if (claim && claim.amount) {
+            try {
+              const claimed = await airdrop.isClaimed(claim.index)
+              if (!claimed) {
+                return { ...claim, airdrop: airdrop.address }
+              }
+              // eslint-disable-next-line
+            } catch {}
+          }
+        }),
+      )
+      return claims.filter(claim => claim)
+    }
+  }
+
+  getClaimAmount = async (address: Maybe<string>) => {
+    const amount = new BigNumber('0')
+    if (this.airdrops) {
+      // get claims across all airdrops
+      const claims = await this.getClaims(address)
+      // aggregate claim amounts
+      return claims?.reduce((totalAmount, claim) => {
+        if (claim) {
+          return totalAmount.add(new BigNumber(claim.amount))
+        }
+        return totalAmount
+      }, amount)
+    }
+    return amount
+  }
+
+  encodeClaims = async (address: string): Promise<Transaction[]> => {
+    const transactions: Transaction[] = []
+    if (this.airdrops) {
+      const claims = await this.getClaims(address)
+      if (claims) {
+        return claims?.map(claim => {
+          const airdropInterface = new utils.Interface(airdropAbi)
+          const data = airdropInterface.functions.claim.encode([
+            claim.index,
+            claim.recipient,
+            claim.amount,
+            claim.proof,
+          ])
+          return {
+            to: claim.airdrop,
+            data,
+          }
+        })
+      }
+    }
+    return transactions
```

Comment:
Wouldn't this always be an empty array?

Reply:
Oh I see it's supposed to be an empty array in this case. In that case we should add a `if (transactions) this.execTransactions...` in `cpk.claimAirdrop`